script/tool/lib/src/common/repository_package.dart

```diff
@@ -132,6 +132,12 @@ class RepositoryPackage {
       !isPlatformInterface &&
       directory.basename != directory.parent.basename;
 
+  /// True if the package is an example for another package.
```

Comment:
It's only slightly divergent from [the official layout docs](https://dart.dev/tools/pub/package-layout#examples). Technically the layout docs would allow for arbitrarily deep nesting of examples within an `example` directory. But in practice, the only cases we have across either repo is a single example called `example`, or an `example` directory directly containing each example.

We can always make this logic more complex if we ever actually have a package with examples three or more levels deep.